rllib/offline/estimators/direct_method.py

```diff
@@ -96,102 +35,62 @@ def __init__(
-        estimates = []
+        estimates = {"v_behavior": [], "v_target": [], "v_gain": []}
         # Split data into train and test batches
-        for train_episodes, test_episodes in train_test_split(
-            batch,
-            self.train_test_split_val,
-            self.k,
-        ):
-
-            # Train Q-function
-            if train_episodes:
-                # Reinitialize model
-                self.model.reset()
-                train_batch = SampleBatch.concat_samples(train_episodes)
-                losses = self.train(train_batch)
-                self.losses.append(losses)
-
-            # Calculate direct method OPE estimates
-            for episode in test_episodes:
-                rewards = episode["rewards"]
-                v_old = 0.0
-                v_new = 0.0
-                for t in range(episode.count):
-                    v_old += rewards[t] * self.gamma ** t
-
-                init_step = episode[0:1]
-                init_obs = np.array([init_step[SampleBatch.OBS]])
-                all_actions = np.arange(self.policy.action_space.n, dtype=float)
-                init_step[SampleBatch.ACTIONS] = all_actions
-                action_probs = np.exp(self.action_log_likelihood(init_step))
-                v_value = self.model.estimate_v(init_obs, action_probs)
-                v_new = convert_to_numpy(v_value).item()
-
-                estimates.append(
-                    OffPolicyEstimate(
-                        self.name,
-                        {
-                            "v_old": v_old,
-                            "v_new": v_new,
-                            "v_gain": v_new / max(1e-8, v_old),
-                        },
-                    )
-                )
+        for episode in batch.split_by_episode():
+            rewards = episode["rewards"]
+            v_behavior = 0.0
+            v_target = 0.0
+            for t in range(episode.count):
+                v_behavior += rewards[t] * self.gamma ** t
+
+            init_step = episode[0:1]
+            v_target = self.model.estimate_v(init_step)
+            v_target = convert_to_numpy(v_target).item()
+
+            estimates["v_behavior"].append(v_behavior)
+            estimates["v_target"].append(v_target)
+            estimates["v_gain"].append(v_target / max(v_behavior, 1e-8))
+        estimates["v_behavior_std"] = np.std(estimates["v_behavior"])
+        estimates["v_behavior"] = np.mean(estimates["v_behavior"])
+        estimates["v_target_std"] = np.std(estimates["v_target"])
+        estimates["v_target"] = np.mean(estimates["v_target"])
+        estimates["v_gain_std"] = np.std(estimates["v_gain"])
+        estimates["v_gain"] = np.mean(estimates["v_gain"])
         return estimates
 
     @override(OffPolicyEstimator)
-    def train(self, batch: SampleBatchType):
-        return self.model.train_q(batch)
+    def train(self, batch: SampleBatchType) -> Dict[str, Any]:
+        if isinstance(batch, MultiAgentBatch):
```

Comment:
**issue**: what are you not using your own checker method? if `action_prob` is an issue break down the function further so that you can reuse the error checking message.

Reply:
Done